src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,252 @@
+import java.util.Random;
+import java.io.FileWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.io.IOException;
+import java.io.File;
+import duke.task.Task;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+public class Duke {
+
+    private static final ArrayList<Task> tasks = new ArrayList<>();
+    private static final String FILE_SEPARATOR = "CHOPCHOP";
+
+    private static void printSeparator() {
+        for(int i = 0; i < 60; i++) {
+            System.out.print('-');
+        }
+        System.out.print('\n');
+    }
+
+    private static void exitMethod() {
+        System.out.print("My cover's blown!\n");
+        printSeparator();
+    }
+
+    /**
+     * Echoes the user input with random upper and lower case for mockery.
+     *
+     * @param line User input.
+     */
+    private static void mockEcho(String line) {
+        Random rd = new Random();
+        for(int i = 0; i < line.length(); i++) {
+            if(rd.nextBoolean()) {
+                System.out.print(Character.toUpperCase(line.charAt(i)));
+            } else {
+                System.out.print(Character.toLowerCase(line.charAt(i)));
+            }
+        }
+        System.out.print('\n');
+    }
+
+    private static void printTasks() throws DukeException{
+        if (tasks.size() < 1) {
+            throw new DukeException();
+        } else {
+            System.out.print("There are " + tasks.size() + " tasks in your list:\n");
+            for(int i = 0; i < tasks.size(); i++) {
+                System.out.print((i+1) + "." + tasks.get(i) + '\n');
+            }
+        }
+    }
+
+    private static void addTask(Task t) throws DukeException{
+        if (t.getDescription().isEmpty()) {
+            throw new DukeException();
+        } else {
+            tasks.add(t);
+            System.out.print("Got it. I've added this task:\n" + t.toString() + '\n');
+        }
+    }
+
+    private static void deleteTask(int taskIndex) throws DukeException{
+        try {
+            System.out.print("Got it. I've deleted this task:\n" + tasks.get(taskIndex).toString() + '\n');
+            tasks.remove(taskIndex);
+        } catch (Exception e) {
+            throw new DukeException();
+        }
+    }
+
+    private static void markIndexDone(int taskIndex) throws DukeException{
+        try {
+            tasks.get(taskIndex).setDone();
+        } catch (Exception e) {
+            throw new DukeException();
+        }
+    }
+
+    private static void printHelp() {
+        System.out.print("I can remember your tasks for you!\n\n" +
+                "Available commands:\n" +
+                "\ttodo <description>\n" +
+                "\tdeadline <description> /by <time due>\n" +
+                "\tevent <description> /at <time occurring>\n" +
+                "\tlist\n" + "\tdone <task index>\n" +
+                "\tdelete <task index>\n");
+    }
+
+    private static void listMode() {
```

Comment:
This method is [long](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods), which could afford some revision. Perhaps abstract some details away?

Reply:
thank you for the comment, Stephanie. May I know how I could abstract the exception handling away? Should I include them within the method which caused the exception? Because currently I'm throwing the exception from within those methods and catching it to print an error message outside. I think that is what is causing the method to be so long.

Reply:
Hi jinwei, you may want to consider some options: put the try catch block in the method where the exception is likely to occur. Or throwing those exceptions from the method you think the exception might occur. See checked exceptions: [https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/cppToJava.html#how-to-use-exceptions](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/cppToJava.html#how-to-use-exceptions).